app/grandchallenge/submission_conversion/models.py

```diff
@@ -92,49 +87,33 @@ def _copy_output_files(self, *, container, base_dir: Path):
         )
 
         if self.__was_unzipped:
+            with TemporaryDirectory() as tmpdir:
+                input_files = set()
 
-            # Create the upload session but do not save it until we have the
-            # files
-            upload_session = RawImageUploadSession(annotationset=annotationset)
-
-            images = []
-
-            for file in output_files:
-                new_uuid = uuid.uuid4()
+                for file in output_files:
+                    tmpfile = Path(
+                        safe_join(tmpdir, file.relative_to(base_dir))
+                    )
+                    tmpfile.parent.mkdir(parents=True, exist_ok=True)
 
```

Comment:
This is the exact same code that is used in algorithm/models.py, correct? If so, could it not be consolidated into a single utility function?

Reply:
This will eventually be consolidated once the Component Interfaces are used for real, rather than the loose API we have now.